### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by xmlking** [grpc/grpc-kotlin#259] [Ideas]
Based on this benchmark review , are there any performance optimizations we can tweak? 

https://www.reddit.com/r/java/comments/n75pa0/comment/gxcl2hq?utm_medium=android_app&utm_source=share&context=3

**Comment by jamesward** [comment]
Yeah, this needs some investigation.
Results:
https://github.com/LesnyRumcajs/grpc_bench/wiki/2021-04-13-bench-results

Kotlin code: https://github.com/LesnyRumcajs/grpc_bench/tree/master/kotlin_grpc_bench
Java code: https://github.com/LesnyRumcajs/grpc_bench/blob/master/java_grpc_sgc_bench/src/main/java/io/grpc/examples/helloworld/HelloWorldServer.java

**Reply by l-monninger** [reply]
Any idea why the Kotlin memory usage is often close to the highest?
